src/functions.php

```diff
@@ -635,3 +649,18 @@ function select_server_for_aggregate_write_stage(Manager $manager, array &$optio
 
     return $server;
 }
+
+/**
+ * Performs server selection for a write operation.
+ *
+ * The pinned server for an active transaction takes priority, followed by an
+ * operation-level read preference, followed by a primary read preference. This
+ * is similar to select_server() except that it ignores a read preference from
+ * an active transaction's options.
+ *
+ * @internal
+ */
+function select_server_for_write(Manager $manager, array $options): Server
```

Comment:
Quoting Transactions: readPreference](https://github.com/mongodb/specifications/blob/master/source/transactions/transactions.rst#readpreference):

> The read preference to use for all read operations in this transaction.

The `select_server_for_write()` method here allows us to avoid considering a transaction's `readPreference` option for a known write command. I intentionally chose _not_ to use this method for read methods that are typically run on primaries (e.g. collection and database enumeration). And it's not used for `Database::command()` sinec that's considered a read operation (see: [runCommand spec](https://github.com/mongodb/specifications/blob/master/source/run-command/run-command.rst#readpreference)).

It's clear we don't have comprehensive test coverage for this across _all_ operations (for example, there's no test that tries to list databases within a transaction), but I'm glad that I was alerted to this oversight in some failing [legacy transaction spec tests](https://spruce.mongodb.com/version/6515c8f8d1fe07b7c48b1f74/tasks).